File: python/code.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# функция не работает, функция "chr" в python не может иметь отрецательное значение / the function does not work, the "chr" function in python cannot have a negative value
def code(text, password, decode):
        if len(password) == 0:
                exit;
        sign = 1
        if decode:
                sign = -1
        result = ""
        for i in range(0, len(text)):
                result += chr(ord(text[i]) + sign * ord(password[i % len(password) + 1]))
        return result

File: php/code.php
function code($text, $password, $decode) {
    $passlen = iconv_strlen($password, 'UTF-8');
    if ($passlen == 0) exit;
    $sign = ($decode) ? -1 : 1;
    $result = '';
    for ($i = 1; $i < iconv_strlen($text, 'UTF-8') + 1; $i++){
        $result = $result.chr(ord(mb_substr($text, ($i - 1), 1, 'UTF-8')) + $sign * ord(mb_substr($password, (($i % $passlen + 1) - 1), 1, 'UTF-8')));
    }
    return $result; 
}
